android/koin-android/src/main/java/org/koin/androidx/viewmodel/dsl/ModuleViewModelOfExt.kt

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2017-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.koin.androidx.viewmodel.dsl
+
+import androidx.lifecycle.ViewModel
+import org.koin.core.module.Module
+import org.koin.core.qualifier.Qualifier
+import org.koin.core.scope.new
+
+/**
+ * Declare a [Module.viewModel] definition by resolving a constructor reference for the dependency.
+ * The resolution is done at compile time by leveraging inline functions, no reflection is required.
+ *
+ * Example:
+ * ```
+ * class ViewModel
+ *
+ * val myModule = module {
+ *   viewModel(::ViewModel)
+ * }
+ * ```
+ *
+ * @author Marcello Galhardo
+ *
+ * @see new
+ */
+inline fun <reified R : ViewModel> Module.viewModelOf(
+    qualifier: Qualifier,
```

Comment:
We can not use the default `null` parameter here, otherwise, the following syntax would not work: `viewModelOf(::MyViewModel)` as the `::MyViewModel` constructor would resolve on the `qualifier` and not on `constructor`.